container/crio/client.go

```diff
@@ -117,21 +133,49 @@ func (c *crioClientImpl) ContainerInfo(id string) (*ContainerInfo, error) {
 	if err != nil {
 		return nil, err
 	}
-	resp, err := c.client.Do(req)
+	var resp *http.Response
+	cInfo := ContainerInfo{}
+	if c.dualStack != dualStackSupported {
+		resp, err := c.client.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+
+		// golang's http.Do doesn't return an error if non 200 response code is returned
+		// handle this case here, rather than failing to decode the body
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("Error finding container %s: Status %d returned error %s", id, resp.StatusCode, resp.Body)
+		}
+
+		if err := json.NewDecoder(resp.Body).Decode(&cInfo); err != nil {
+			return nil, err
+		}
+		if len(cInfo.IP) > 0 {
+			c.dualStack = dualStackNotSupported
+			return &cInfo, nil
+		}
+	}
+	// try to obtain IP address by parsing ip_addresses JSON field
+	if resp != nil {
+		resp.Body.Close()
+	}
+	resp, err = c.client.Do(req)
 	if err != nil {
 		return nil, err
 	}
 	defer resp.Body.Close()
 
-	// golang's http.Do doesn't return an error if non 200 response code is returned
-	// handle this case here, rather than failing to decode the body
 	if resp.StatusCode != http.StatusOK {
 		return nil, fmt.Errorf("Error finding container %s: Status %d returned error %s", id, resp.StatusCode, resp.Body)
 	}
-
-	cInfo := ContainerInfo{}
-	if err := json.NewDecoder(resp.Body).Decode(&cInfo); err != nil {
+	cInfoDualStack := containerInfoDualStack{}
+	if err := json.NewDecoder(resp.Body).Decode(&cInfoDualStack); err != nil {
 		return nil, err
 	}
+	if len(cInfoDualStack.IPs) > 0 {
+		cInfo.IP = cInfoDualStack.IPs[0]
+		c.dualStack = dualStackSupported
```

Comment:
Do we use this information anywhere?  What is the purpose of storing and adding it?

Reply:
Dropped